### DISCUSSION TITLE
markdown visual editor sentence wrap in figure captions

### DISCUSSION THREAD

**Discussion by dhimmel** [quarto-dev/quarto-cli#11668] [Feature Requests]
As someone who tracks manuscript source with git, I'm happy to see the editor option:

```yaml
editor:
  markdown:
    wrap: sentence
```

It would be nice if sentence wrap could be applied in figure captions as well like:

```md
![
  **My figure caption**:
  Captions in academic writing are often long.
  One sentence/phrase per line is the best.
  In semantic linefeeds, we trust.
](path-to-figure.svg){#fig-my-figure}
```

Currently the VSCode editor converts to:

```md
![**My figure caption**: Captions in academic writing are often long. One sentence/phrase per line is the best. In semantic linefeeds, we trust.](path-to-figure.svg){#fig-my-figure}
```

A somewhat related request is an option to only wrap sections of text that the user edits in the visual editor as opposed to wrapping everything. Without this, it's hard to use the visual editor to make small changes without blowing up the diff.

**Comment by mcanouil** [comment]
The repository for the VSCode extension is https://github.com/quarto-dev/quarto

I don't think this enhancement request was already made, but please check the opened issues first before opening one.

Note that `editor.markdown` is not for the Visual Editor, it's for the markdown editor.
You won't have line wrapping in the Visual Editor with the setting, you'll have that when looking at the source editor (markdown).

Another note `![](){}` is an inline syntax in Pandoc's view (used behind the scene by the Visual Editor to reformat the document).
Allowing line wrap/break would be very tricky inside this syntax especially if you start to add cross-references in it with `[-@ref]` syntax. See below for a "better" alternative.

---

Alternatively, you could use the cross-reference div syntax which in my opinion is better suited for long caption. Using this syntax should trigger the visual editor line wrapping I believe.

````qmd
::: {#fig-my-figure}
![](path-to-figure.svg)

**My figure caption**: Captions in academic writing are often long. One sentence/phrase per line is the best. In semantic linefeeds, we trust.
:::
````

**Reply by dhimmel** [reply]
Ah nice tip about the [Figure Div](https://quarto.org/docs/authoring/figures.html#figure-divs) method.

> You can treat any markdown content you want as a figure by enclosing it in Pandoc div block with an identifier prefaced with `#fig-`

It's nice that these captions are still put in `<figcaption>` and work with figure panels. This is a sufficient solution for my issue and I agree its a clearer syntax for long captions.

> Note that `editor.markdown` is not for the Visual Editor, it's for the markdown editor.

Hmm. It seems to do what I want which is to [instruct](https://quarto.org/docs/visual-editor/vscode/#line-wrapping) the VSCode visual editor to save the markdown using one sentence per line. In the visual editor rendered view, the newlines won't appear of course.

> The repository for the VSCode extension is https://github.com/quarto-dev/quarto. I don't think this enhancement request was already made, but please check the opened issues first before opening one.

It seems that there are multiple visual editors, one for VS Code, one for RStudio. Are the editor settings and behaviors synced between the multiple visual editors? In other words, does a request like mine need to be filed in a more general location?

**Reply by mcanouil** [reply]
> It seems that there are multiple visual editors, one for VS Code, one for RStudio. Are the editor settings and behaviors synced between the multiple visual editors? In other words, does a request like mine need to be filed in a more general location?

There is only one which is used by RStudio AND VSCode extension

**Reply by cscheid** [reply]
> In other words, does a request like mine need to be filed in a more general location?

Feel free to start at `quarto-dev/quarto` and, if needed, we'll transfer the issue. `quarto-dev/quarto` is a monorepo that currently hosts most of the code for Quarto's editor support in RStudio, as well as VS code + Positron.
